File: c++/leetcode/476. Number Complement.cpp
class Solution
{
public:
  int findComplement(int num)
  {
    int dupe = num;
    int mask = 0;

    while (dupe != 0)
    {
      dupe >>= 1;
      mask = (mask << 1) | 1;
    }

    return num ^ mask;
  }
};

File: javascript/leetcode/476. Number Complement.js
/**
 * @param {number} num
 * @return {number}
 */
var findComplement = function (num) {
  let dupe = num;
  let mask = 0;

  while (dupe != 0) {
    dupe >>= 1;
    mask = (mask << 1) | 1;
  }

  return num ^ mask;
};
